benchmarks/benchmark_shared.c

```diff
@@ -192,3 +192,92 @@ int bandwidth(void)
 
 	return 0;
 }
+
+int bandwidth_rma(enum ft_rma_opcodes rma_op, struct fi_rma_iov *remote)
+{
+	int ret, i, j;
+
+	ret = ft_sync();
+	if (ret)
+		return ret;
+
+	if (!ctx_arr) {
+		return -FI_ENOMEM;
+	}
+
+	for (i = 0; i < opts.iterations + opts.warmup_iterations; i++) {
+		if (i == opts.warmup_iterations)
+			ft_start();
+
+		for(j = 0; j < opts.window_size; j++) {
+			switch (rma_op) {
+			case FT_RMA_WRITE:
+				if (opts.transfer_size < fi->tx_attr->inject_size) {
+					ret = ft_post_rma_inject(FT_RMA_WRITE, ep,
+							opts.transfer_size, remote);
+				} else {
+					ret = ft_post_rma(rma_op, ep, opts.transfer_size,
+							remote,	&ctx_arr[j]);
+				}
+				break;
+			case FT_RMA_WRITEDATA:
+				if (!opts.dst_addr) {
+					ret = ft_post_rx(ep, 0, &ctx_arr[j]);
+					if (ret)
+						return ret;
+				} else {
+					if (opts.transfer_size < fi->tx_attr->inject_size) {
+						ret = ft_post_rma_inject(FT_RMA_WRITEDATA,
+								ep,
+								opts.transfer_size,
+								remote);
+					} else {
+						ret = ft_post_rma(FT_RMA_WRITEDATA,
+								ep,
+								opts.transfer_size,
+								remote,	&ctx_arr[j]);
+					}
+				}
+				break;
+			case FT_RMA_READ:
+				ret = ft_post_rma(FT_RMA_READ, ep, opts.transfer_size,
+						remote,	&ctx_arr[j]);
+				break;
+			default:
+				FT_ERR("Unknown RMA op type\n");
+				return EXIT_FAILURE;
+			}
+			if (ret)
+				return ret;
+		}
+
+		if (rma_op == FT_RMA_WRITEDATA) {
+			if (!opts.dst_addr) {
+				/* rx_seq is always one ahead */
+				ret = ft_get_rx_comp(rx_seq - 1);
+				if (ret)
+					return ret;
+				ret = ft_tx(ep, 4);
+				if (ret)
+					return ret;
+			} else {
+				ret = ft_rx(ep, 4);
+				if (ret)
+					return ret;
+			}
+		} else {
+			ret = ft_get_tx_comp(tx_seq);
+			if (ret)
+				return ret;
+		}
```

Comment:
Okay, I guess the if (ret) following the ft_get_rx_comp should still be there to handle its failure.